### DISCUSSION TITLE
tbl_add_reverse_lookup has been deprecated, that is why, and what is the substitutes?

### DISCUSSION THREAD

**Discussion by nolleh** [neovim/neovim#29065] [Q&A] [answered]
as described in title, 
`vim.tbl_add_reverse_lookup` has been deprecated in new version of neovim.

1. is there reason that deprecated ? 
 - by guessing, it looks like not frequently used? 

2. and if there is substitution, can you inform us? 
3. and if there is substitution, can you add the suggestion to vim.deprecated message ?

**Comment by clason** [comment, marked as answer]
1. The reason is that it was _bad_: Shoving the reverse lookup into the same table just for the sake of saving a single table is not worth the headache it introduces with for type instability and bugs. (Hitting those in Neovim were exactly what triggered the removal -- if _we_ can't use it correctly, we can't in good conscience support it as an official API for others.)

2. The answer is simple: just create a separate lookup table _if you really need one_; StackOverflow or ChatGPT will be more than happy to give you the simple code snippet for it.

3. And since there is no substitution ("just don't do this, period"), there is no alternative to suggest.

**Reply by nolleh** [reply]
I understood, and agreed that the reason why that is should be removed.   
thanx for your quick and kind explanation! :)

**Comment by wookayin** [comment]
See also #27639
